core/controllers/library.py

```diff
@@ -33,9 +33,30 @@
 import utils
 
 
-def get_matching_activity_dicts(query_string, search_cursor):
-    """Given a query string and a search cursor, returns a list of activity
-    dicts that satisfy the search query.
+def get_matching_activity_dicts(
+        query_string, categories, language_codes, search_cursor):
+    """Given the details of a query and a search cursor, returns a list of
+    activity dicts that satisfy the query.
+
+    Args:
+        query_string: str. The search query string (this is what the user
+            enters).
+        categories: list(str). The list of categories to query for. If it is
+            empty, no category filter is applied to the results. If it is not
+            empty, then a result is considered valid if it matches at least one
+            of these categories.
+        language_codes: list(str). The list of language codes to query for. If
+            it is empty, no language code filter is applied to the results. If
+            it is not empty, then a result is considered valid if it matches at
+            least one of these language codes.
+        search_cursor: str or None. Cursor indicating where, in the list of
+            search results, to start the search from.
+
+    Returns:
+        tuple. A tuple consisting of two elements:
+            - list(dict). Each element in this list is a collection or
+                exploration summary dict, representing a search result.
+            - str. The cursor from which to start the next search.
```

Comment:
The cursor is only used for exploration search, right?

Reply:
Yep, updated docstring. Thanks!